src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/Fixers/SourceGeneratorAttributeAnalyzerFix.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.Fixers
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic, Name = nameof(SourceGeneratorAttributeAnalyzerFix))]
+    [Shared]
+    public sealed class SourceGeneratorAttributeAnalyzerFix : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(
+            DiagnosticIds.MissingSourceGeneratorAttributeId);
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var root = await document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var node = root.FindNode(context.Span, getInnermostNodeForTie: true);
+
+            if (node is null)
+            {
+                return;
+            }
+
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                context.RegisterCodeFix(
+                    CodeAction.Create(
+                        CodeAnalysisDiagnosticsResources.AddGeneratorAttribute,
+                        c => FixDocumentAsync(document, node, c),
+                        equivalenceKey: nameof(SourceGeneratorAttributeAnalyzerFix)),
```

Comment:
I think it doesn't matter, but this should be more consistent with other usages of equivalence key.

```suggestion
                        equivalenceKey: nameof(CodeAnalysisDiagnosticsResources.AddGeneratorAttribute)),
```

Reply:
hehe, I pulled this from another fixer :P if this is the normal key that works for me